apps-rendering/src/client/newsletterEmbeds.ts

```diff
@@ -0,0 +1,62 @@
+import { isObject } from 'lib';
+
+const allowedOrigins = ['https://www.theguardian.com'];
+
+export default (): Promise<void> => {
+	const allIframes: HTMLIFrameElement[] = [].slice.call(
+		document.querySelectorAll('.email-sub__iframe'),
+	);
+
+	// Tell the iframes to resize once this script is loaded
+	// Otherwise, earlier resize events might be missed
+	// So we don't have to load this script as a priority on each load
+	allIframes.forEach((iframe) => {
+		iframe.contentWindow?.postMessage(
+			'resize',
+			'https://www.theguardian.com',
+		);
+	});
+
+	window.addEventListener('message', (event) => {
+		if (!allowedOrigins.includes(event.origin)) return;
+
+		const iframes: HTMLIFrameElement[] = allIframes.filter((i) => {
+			try {
+				if (i.contentWindow && event.source) {
```

Comment:
Small suggestion: can I recommend explicitly checking both against `null` here? E.g. `i.contentWindow !== null`. I think it's easier to understand at a glance what the values can be and what you're checking when reading the code, and implicit conversions can cause unintended consequences at times.